documentation/ADRs/027_ensmeble_reconciliation.md

```diff
@@ -0,0 +1,47 @@
+Ensemble reconcilation
+
+| ADR Info            | Details                 |
+|---------------------|-------------------------|
+| Subject             | Ensemble reconciliation |
+| ADR Number          | 027                     |
+| Status              | proposed                |
+| Author              | Jim                     |
+| Date                | 01/11/2024              |
+
+## Context
+The notebook-based views3/fatalities002 pipeline generates a cm and a pgm ensemble. It was found that the pgm ensemble suffered from what might be termed normalisation issues, in that the peak and total numbers of fatalities forecast at pgm level are clearly too low. In particular, summing forecast fatalities over the pg cells belonging to a given country, a dcomapring to the fatalities forecast for the same country at cm level almost always gives the result that the summed pgm values are significantly - often an order of magnitude - lower.
+As a quick fix, therefore, a reconciliation function was created which accepts a pgm forecast dataframe and a cm forecast dataframe, fetches via viewser a pgm->cm mapping, computes for every country for every month the sum over its constituent pg cells, and renormalises the pgm forecasts for those cells so that the sum matches the cm-level forecast. A check is performed which ensures that the set of months in the two input dfs is the same.
+The reconciliation will be applied to every pgm-level constituent model from which the pgm ensemble is built.
```

Comment:
I would emphasize that this is a known issue with legacy models that were adapted in various forms from the old pipeline. Although some hyper-parameters might help mitigate these issues, the challenges are inherent to the models' architecture and loss functions.

bla bla bla

Going forward, the explicit goal for all model development efforts is to design architectures, loss functions, optimization routines, sampling strategies, and other methods that address these issues.

bla bla bla

We will likely continue reconciliation as a stable approach, but its purpose should be to maintain strict consistency between CM-level and aggregated PGM data—not to systematically up-bias PGM models that underestimate conflict fatalities. This underestimation is fundamentally a modeling issue, not a reconciliation problem.

Reply:
Implemented :)

Reply:
Noise